ngraph/core/include/ngraph/type/element_type.hpp

```diff
@@ -91,11 +91,12 @@ namespace ngraph
             size_t bitwidth() const;
             // The name of this type, the enum name of this type
             const std::string& get_type_name() const;
+            bool operator==(const Type_t& other) const;
```

Comment:
Have you try to remove all of comparing functions and let allow compiler to use cast to enum (l:127) to do comparators.